frontend/src/components/topicsPage/TreeTab.js

```diff
@@ -1,21 +1,146 @@
-import React from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useState, useEffect } from 'react';
 import {
-  Container, Row, Col,
+  Container,
+  Box,
+  Heading,
+  Loader,
+  InfoIcon,
+  Palette,
+  FormField,
 } from 'wix-style-react';
-import Tree from './tree/Tree';
-import data from './tree/data';
+import { useSelector, useDispatch } from 'react-redux';
+import TreeContainer from './tree/TreeContainer';
 import SelectTreeForm from './SelectTreeForm';
+import {
+  TOPICS_TREE_HEIGHT, TOPICS_TREE_WIDTH, TREE_NODE_LEARNED_COLOR, TREE_NODE_NOT_PLANNED_COLOR, TREE_NODE_PLANNED_COLOR, TREE_NODE_BADGE_COLOR,
+} from '../../constants/Styling';
+import {
+  PERSONAL, MY_TEAM, MY_SUBORDINATES, SINGLE_SUBORDINATE, SINGLE_TEAM,
+} from '../../constants/TreeTypes';
+import {
+  getPersonalTree, getMyTeamTree, getMySubordinatesTree, getSingleSubordinateTree, getSingleTeamTree,
+} from '../../state/actions';
+import { LOADING_PERSONAL_TREE, FETCH_PERSONAL_TREE_SUCCEEDED } from '../../constants/PersonalTreeStatus';
+import { LOADING_MY_TEAM_TREE, FETCH_MY_TEAM_TREE_SUCCEEDED } from '../../constants/MyTeamTreeStatus';
+import { LOADING_MY_SUBORDINATES_TREE, FETCH_MY_SUBORDINATES_TREE_SUCCEEDED } from '../../constants/MySubordinatesTreeStatus';
+import { LOADING_SINGLE_SUBORDINATE_TREE, FETCH_SINGLE_SUBORDINATE_TREE_SUCCEEDED } from '../../constants/SingleSubordinateTreeStatus';
+import { LOADING_SINGLE_TEAM_TREE, FETCH_SINGLE_TEAM_TREE_SUCCEEDED } from '../../constants/SingleTeamTreeStatus';
 
 const TreeTab = () => {
+  const initialTree = {
+    name: 'Topics',
+    children: [],
+  };
+
+  const [tree, setTree] = useState(initialTree);
+  const [treeName, setTreeName] = useState('');
+
+  const dispatch = useDispatch();
+
+  const { tree: personalTree, status: personalTreeStatus } = useSelector(state => state.personalTree);
+  const { tree: myTeamTree, status: myTeamTreeStatus } = useSelector(state => state.myTeamTree);
+  const { tree: mySubordinatesTree, status: mySubordinatesTreeStatus } = useSelector(state => state.mySubordinatesTree);
+  const { tree: singleSubordinateTree, status: singleSubordinateTreeStatus } = useSelector(state => state.singleSubordinateTree);
+  const { tree: singleTeamTree, status: singleTeamTreeStatus } = useSelector(state => state.singleTeamTree);
+
+  const reformatDataToTree = data => {
+    const newTree = {
+      name: 'Topics',
+      children: data,
+    };
+    setTree(newTree);
+  };
+
+  const isLoading = personalTreeStatus === LOADING_PERSONAL_TREE || myTeamTreeStatus === LOADING_MY_TEAM_TREE || mySubordinatesTreeStatus === LOADING_MY_SUBORDINATES_TREE || singleSubordinateTreeStatus === LOADING_SINGLE_SUBORDINATE_TREE || singleTeamTreeStatus === LOADING_SINGLE_TEAM_TREE;
```

Comment:
make it multiline

Reply:
ALSO, more importantly, change it to
`const isLoading = personalTreeStatus !== FETCH_PERSONAL_TREE_SUCCEEDED || ...;`
Or, write a `const hasFailed = personalTreeStatus === FETCH_PERSONAL_TREE_FAILED || ...;` and then display something in the component, e.g. "Couldn't display tree"